### PR TITLE
Switch to using the RCUTILS_* logging macros.

### PR DESCRIPTION
**Pull request by clalancette**
Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

### CODE REVIEW

**Comment by clalancette** [comment]
CI:

* Turtlebot [Build Status](http://ci.ros2.org/job/ci_turtlebot-demo_linux/98/)

**Comment by mikaelarguedas** [comment]
Commenting here but (may) apply to all related PRs: please remove trailing newlines as well as trailing semi-colons in macro invocations when invoking directly RCUTILS macros

**Comment by clalancette** [comment]
> Commenting here but (may) apply to all related PRs: please remove trailing newlines as well as trailing semi-colons in macro invocations when invoking directly RCUTILS macros

Sorry, that was me just being buggy.  This should be fixed now.

File: depthimage_to_pointcloud2/src/depthimage_to_pointcloud2_node.cpp [hunk 1]
```diff
@@ -64,7 +65,7 @@ static void depthCb(const sensor_msgs::msg::Image::SharedPtr image)
   } else if (image->encoding == sensor_msgs::image_encodings::TYPE_32FC1) {
     depthimage_to_pointcloud2::convert<float>(image, cloud_msg, model);
   } else {
-    fprintf(stderr, "Depth image has unsupported encoding [%s]\n", image->encoding.c_str());
+    RCUTILS_LOG_WARN("Depth image has unsupported encoding [%s]", image->encoding.c_str())
```
Comment (mikaelarguedas) [inline depthimage_to_pointcloud2/src/depthimage_to_pointcloud2_node.cpp]: We could use a throttled logging here to match the original code.

File: depthimage_to_pointcloud2/src/depthimage_to_pointcloud2_node.cpp [hunk 2]
```diff
@@ -37,7 +38,7 @@ static void depthCb(const sensor_msgs::msg::Image::SharedPtr image)
 
   if (nullptr == g_cam_info) {
     // we haven't gotten the camera info yet, so just drop until we do
-    fprintf(stderr, "No camera info, skipping point cloud conversion\n");
+    RCUTILS_LOG_WARN("No camera info, skipping point cloud conversion")
```
Comment (mikaelarguedas) [inline depthimage_to_pointcloud2/src/depthimage_to_pointcloud2_node.cpp:41]: I think this (and the next one below) should be ERROR and not WARN given that they result in the function returning without publishing the pointcloud

File: turtlebot2_drivers/src/kobuki_node.cpp [hunk 3]
```diff
@@ -98,9 +99,9 @@ int main(int argc, char * argv[])
   g_max_vyaw = 1.0;
   node->get_parameter("max_vyaw", g_max_vyaw);
 
-  printf("device_port: %s\n", parameters.device_port.c_str());
-  printf("max_vx: %f\n", g_max_vx);
-  printf("max_vyaw: %f\n", g_max_vyaw);
+  RCUTILS_LOG_INFO("device_port: %s", parameters.device_port.c_str())
+  RCUTILS_LOG_INFO("max_vx: %f", g_max_vx)
+  RCUTILS_LOG_INFO("max_vyaw: %f", g_max_vyaw)
```
Comment (mikaelarguedas) [inline turtlebot2_drivers/src/kobuki_node.cpp]: These should be DEBUG IMO

File: turtlebot2_follower/src/follower.cpp [hunk 4]
```diff
@@ -160,7 +163,7 @@ class TurtlebotFollower
     // We're assuming floating point data
     if(depth_msg->encoding != sensor_msgs::image_encodings::TYPE_32FC1)
     {
-      printf("received depth image with unsupported encoding: %s\n", depth_msg->encoding.c_str());
+      ROS_WARN("received depth image with unsupported encoding: %s", depth_msg->encoding.c_str());
```
Comment (mikaelarguedas) [inline turtlebot2_follower/src/follower.cpp]: should have ERROR severity

File: depthimage_to_pointcloud2/src/depthimage_to_pointcloud2_node.cpp (hunk 1 above)
Comment (clalancette) [reply depthimage_to_pointcloud2/src/depthimage_to_pointcloud2_node.cpp]: Will change this.

File: depthimage_to_pointcloud2/src/depthimage_to_pointcloud2_node.cpp (hunk 2 above)
Comment (clalancette) [reply depthimage_to_pointcloud2/src/depthimage_to_pointcloud2_node.cpp:41]: True, but for this one in particular, it happens regularly at startup time.  Sometimes this node comes up before both the depth image and the camera info have been published, so you see this message those times.  At that point it is not an error, but really just informational.  I'm not sure what we should do with that.

File: turtlebot2_drivers/src/kobuki_node.cpp (hunk 3 above)
Comment (clalancette) [reply turtlebot2_drivers/src/kobuki_node.cpp]: Yeah, I'll change that.

File: turtlebot2_follower/src/follower.cpp (hunk 4 above)
Comment (clalancette) [reply turtlebot2_follower/src/follower.cpp]: Sounds good, will change.

File: depthimage_to_pointcloud2/src/depthimage_to_pointcloud2_node.cpp [hunk 5]
```diff
@@ -64,7 +65,7 @@ static void depthCb(const sensor_msgs::msg::Image::SharedPtr image)
   } else if (image->encoding == sensor_msgs::image_encodings::TYPE_32FC1) {
     depthimage_to_pointcloud2::convert<float>(image, cloud_msg, model);
   } else {
-    fprintf(stderr, "Depth image has unsupported encoding [%s]\n", image->encoding.c_str());
+    RCUTILS_LOG_WARN_THROTTLE(RCUTILS_STEADY_TIME, 5000, "Depth image has unsupported encoding [%s]", image->encoding.c_str())
```
Comment (mikaelarguedas) [inline depthimage_to_pointcloud2/src/depthimage_to_pointcloud2_node.cpp]: This line is very long and I think will fail the linters

Comment (clalancette) [reply depthimage_to_pointcloud2/src/depthimage_to_pointcloud2_node.cpp]: Yeah, good point.  Fixed in 5281c00

**Comment by clalancette** [comment]
Just to double-check, here is another CI job: [Build Status](http://ci.ros2.org/job/ci_turtlebot-demo_linux/99/)

File: turtlebot2_follower/src/follower.cpp [hunk 6]
```diff
@@ -226,17 +230,15 @@ class TurtlebotFollower
       x /= n;
       y /= n;
       if(z > max_z_){
-        //ROS_INFO_THROTTLE(1, "Centroid too far away %f, stopping the robot\n%s", z);
-        printf("Centroid too far away %f, stopping the robot\n", z);
+        ROS_INFO_THROTTLE(1, "Centroid too far away %f, stopping the robot\n", z);
```
Comment (mikaelarguedas) [inline turtlebot2_follower/src/follower.cpp]: Copy-n-paste error ? The formatting (`%s`) for z got removed from the logging invocation.

Comment (clalancette) [reply turtlebot2_follower/src/follower.cpp]: That is actually a bug fix.  If you look closely, there is only one argument to the printf-like function, `z`.  In the upstream version of the code, the `%s` doesn't exist; somehow it snuck in during the port.  g++ throws an error with the `%s` in, so I just removed it which makes this look like the upstream code and fixes the compilation.

Comment (mikaelarguedas) [reply turtlebot2_follower/src/follower.cpp]: sounds good, looks like the %s was there at the time the file was ported and that the bugfix from upstream didnt get synced here.
In that case the extra `\n` should be removed here as well as upstream.
Additionally we could provide a reference to the upstream file to make comparing with upstream this more easy in the future

Comment (clalancette) [reply turtlebot2_follower/src/follower.cpp]: Sure, I'll add in a reference to the upstream file.  I'll also open a PR on the upstream to remove the `\n`.

Comment (clalancette) [reply turtlebot2_follower/src/follower.cpp]: https://github.com/turtlebot/turtlebot_apps/pull/165

Comment (mikaelarguedas) [reply turtlebot2_follower/src/follower.cpp]: awesome thanks, good te be merged as soon as the extra `\n` is removed from this file as well

Comment (clalancette) [reply turtlebot2_follower/src/follower.cpp]: Done now in 5ae2daa

**Review by mikaelarguedas — APPROVED** [review]
lgtm

**Comment by clalancette** [comment]
Thanks again!  Merging.

**Pull request merged**